Set RUSTC_BOOTSTRAP=1 in the environment so that stable versions of rustc
allow the use of -Z* compiler options, rather than erroring out with

  error: the option `Z` is only accepted on the nightly compiler

--- a/build/rust/gni_impl/rustc_wrapper.py
+++ b/build/rust/gni_impl/rustc_wrapper.py
@@ -401,6 +401,9 @@ def main():
   # (but do this _after_ the `_SaveRustEnvAndFlags` call above).
   rustc_args += ["--emit", args.emit, "-o", args.o]
 
+  # Debian: Allow -Z* compiler options
+  rustenv['RUSTC_BOOTSTRAP'] = '1'
+
   if args.v:
     print(' '.join(f'{k}={shlex.quote(v)}' for k, v in rustenv.items()),
           args.rustc, shlex.join(rustc_args))
